4. REGIONAL ANALYSIS

1. BUSINESS UNDERSTANDING

Perusahaan ingin memahami sales dari setiap wilayah agar mengetahui :

-> Wilayah yang memiliki pelanggan terbanyak.

-> Wilayah dengan penjualan terbesar

-> Wilayah dengan rata rata penjualan terbesar.

2. DATA UNDERSTANDING

In [39]:
import pandas as pd

In [40]:
df = pd.read_csv(r"D:\PROJECT\Superstore Sales Dataset\Dataset\Superstore Sales Dataset.csv")
df.head()

,Row ID,Order ID,Order Date,Ship Date,Ship Mode,Customer ID,Customer Name,Segment,Country,City,State,Postal Code,Region,Product ID,Category,Sub-Category,Product Name,Sales
0,1,CA-2017-152156,8/11/2017,11/11/2017,Second Class,CG-12520,Claire Gute,Consumer,United States,Henderson,Kentucky,42420.0,South,FUR-BO-10001798,Furniture,Bookcases,Bush Somerset Collection Bookcase,261.9600
1,2,CA-2017-152156,8/11/2017,11/11/2017,Second Class,CG-12520,Claire Gute,Consumer,United States,Henderson,Kentucky,42420.0,South,FUR-CH-10000454,Furniture,Chairs,"Hon Deluxe Fabric Upholstered Stacking Chairs,...",731.9400
2,3,CA-2017-138688,12/6/2017,16/06/2017,Second Class,DV-13045,Darrin Van Huff,Corporate,United States,Los Angeles,California,90036.0,West,OFF-LA-10000240,Office Supplies,Labels,Self-Adhesive Address Labels for Typewriters b...,14.6200
3,4,US-2016-108966,11/10/2016,18/10/2016,Standard Class,SO-20335,Sean O'Donnell,Consumer,United States,Fort Lauderdale,Florida,33311.0,South,FUR-TA-10000577,Furniture,Tables,Bretford CR4500 Series Slim Rectangular Table,957.5775
4,5,US-2016-108966,11/10/2016,18/10/2016,Standard Class,SO-20335,Sean O'Donnell,Consumer,United States,Fort Lauderdale,Florida,33311.0,South,OFF-ST-10000760,Office Supplies,Storage,Eldon Fold 'N Roll Cart System,22.3680


In [41]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9800 entries, 0 to 9799
Data columns (total 18 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Row ID         9800 non-null   int64  
 1   Order ID       9800 non-null   object 
 2   Order Date     9800 non-null   object 
 3   Ship Date      9800 non-null   object 
 4   Ship Mode      9800 non-null   object 
 5   Customer ID    9800 non-null   object 
 6   Customer Name  9800 non-null   object 
 7   Segment        9800 non-null   object 
 8   Country        9800 non-null   object 
 9   City           9800 non-null   object 
 10  State          9800 non-null   object 
 11  Postal Code    9789 non-null   float64
 12  Region         9800 non-null   object 
 13  Product ID     9800 non-null   object 
 14  Category       9800 non-null   object 
 15  Sub-Category   9800 non-null   object 
 16  Product Name   9800 non-null   object 
 17  Sales          9800 non-null   float64
dtypes: float

In [42]:
df.shape

(9800, 18)

In [43]:
print(df.columns)

Index(['Row ID', 'Order ID', 'Order Date', 'Ship Date', 'Ship Mode',
       'Customer ID', 'Customer Name', 'Segment', 'Country', 'City', 'State',
       'Postal Code', 'Region', 'Product ID', 'Category', 'Sub-Category',
       'Product Name', 'Sales'],
      dtype='object')


-> Data set memiliki 9800 data dengan 18 kolom, dengan tipe data yg berbeda terutama untuk kolom Postal Code dan Sales memiliki tipe data float64, yang lain berupa object.

3. DATA PREPARATION

In [44]:
df.isnull().sum()

Row ID            0
Order ID          0
Order Date        0
Ship Date         0
Ship Mode         0
Customer ID       0
Customer Name     0
Segment           0
Country           0
City              0
State             0
Postal Code      11
Region            0
Product ID        0
Category          0
Sub-Category      0
Product Name      0
Sales             0
dtype: int64

In [45]:
df.duplicated().sum()

0

In [46]:
df = df.drop(columns=["Postal Code"])

-> Sesuai dengan pengamatan sebelumnya bahwa data yang memiliki nilai nol hanya di kolom Postal Code dan data kembar tidak ada.

-> Pada analisis kali ini kolom Postal Code di hapus

4. REGIONAL OVERVIEW

In [47]:
total_city = df["City"].nunique()
total_state = df["State"].nunique()
total_region = df["Region"].nunique()

print(f"Kota          : {total_city}")
print(f"Negara Bagian : {total_state}")
print(f"Wilayah       : {total_region}")

Kota          : 529
Negara Bagian : 49
Wilayah       : 4


-> Dari dataset ada kota sebanyak 529, dari 49 negara bagian dan 4 wilayah

5. Sales by Region

In [48]:
region_sales = (
    df.groupby(["Region"]) ["Sales"]
    .sum()
    .sort_values(ascending=False)
    .reset_index()
)

region_order = (
    df.groupby(["Region"]) ["Order ID"]
    .nunique()
    .sort_values(ascending=False)
    .reset_index()
)

region_customer = (
    df.groupby(["Region"]) ["Customer ID"]
    .nunique()
    .sort_values(ascending=False)
    .reset_index()
)



In [49]:
region_all = (
    region_sales
    .merge(region_order, on="Region")
    .merge(region_customer, on="Region")
)

# Menghitung kontribusi (%)
region_all["Contribution (%)"] = (
    region_all["Sales"] / region_all["Sales"].sum() * 100
).round(2)

# Menghitung rata rata pembelian
region_all["Average"] = (
    region_all["Sales"] / region_all["Order ID"]
)
region_all

,Region,Sales,Order ID,Customer ID,Contribution (%),Average
0,West,710219.6845,1587,681,31.40,447.523431
1,East,669518.7260,1369,669,29.60,489.056776
2,Central,492646.9132,1156,626,21.78,426.165150
3,South,389151.4590,810,509,17.21,480.433900


In [50]:
region_sales_display = region_all.copy()

region_sales_display["Sales"] = region_sales_display["Sales"].map("${:,.2f}".format)

region_sales_display["Average"] = region_sales_display["Average"].map("${:,.2f}".format)

region_sales_display

,Region,Sales,Order ID,Customer ID,Contribution (%),Average
0,West,"$710,219.68",1587,681,31.40,$447.52
1,East,"$669,518.73",1369,669,29.60,$489.06
2,Central,"$492,646.91",1156,626,21.78,$426.17
3,South,"$389,151.46",810,509,17.21,$480.43


-> Region dengan jumlah penjualan terbanyak, Order terbanyak, dan Kontribusi terbesar ada di region West dengan nilai Sales $710,219.68, order sebanyak 1587, dan kontribusi sebesar 31.40% tidak hanya itu saja region West memiliki jumlah customer terbanyak diantara region yang lainnya.

-> Region dengan jumlah penjualan terkecil, Order terkecil, dan Kontribusi terkecil ada di region South dengan nilai Sales $389,151.46, order sebanyak 810, dan kontribusi sebesar 17.21%

6. Sales by State

In [51]:
state_sales = (
    df.groupby(["Region","State"]) ["Sales"]
    .sum()
    .sort_values(ascending=False)
    .reset_index()
)

state_order = (
    df.groupby(["Region","State"]) ["Order ID"]
    .nunique()
    .sort_values(ascending=False)
    .reset_index()
)

state_all = pd.merge(
    state_sales,
    state_order,
    on=["Region", "State"]
)

# Menghitung kontribusi (%)
state_all["Contribution (%)"] = (
    state_all["Sales"] / state_all["Sales"].sum() * 100
).round(2)

state_all.head(10)

,Region,State,Sales,Order ID,Contribution (%)
0,West,California,446306.4635,1002,19.73
1,East,New York,306361.1470,547,13.55
2,Central,Texas,168572.5322,480,7.45
3,West,Washington,135206.8500,254,5.98
4,East,Pennsylvania,116276.6500,285,5.14
5,South,Florida,88436.5320,197,3.91
6,Central,Illinois,79236.5170,270,3.50
7,Central,Michigan,76136.0740,116,3.37
8,East,Ohio,75130.3500,227,3.32
9,South,Virginia,70636.7200,115,3.12


In [52]:
state_display = state_all.copy()

state_display["Sales"] = state_display["Sales"].map("${:,.2f}".format)

state_display.head(10)

,Region,State,Sales,Order ID,Contribution (%)
0,West,California,"$446,306.46",1002,19.73
1,East,New York,"$306,361.15",547,13.55
2,Central,Texas,"$168,572.53",480,7.45
3,West,Washington,"$135,206.85",254,5.98
4,East,Pennsylvania,"$116,276.65",285,5.14
5,South,Florida,"$88,436.53",197,3.91
6,Central,Illinois,"$79,236.52",270,3.50
7,Central,Michigan,"$76,136.07",116,3.37
8,East,Ohio,"$75,130.35",227,3.32
9,South,Virginia,"$70,636.72",115,3.12


In [53]:
state_display.tail(10)

,Region,State,Sales,Order ID,Contribution (%)
39,West,New Mexico,"$4,783.52",22,0.21
40,Central,Iowa,"$4,443.56",16,0.20
41,West,Idaho,"$4,382.49",11,0.19
42,Central,Kansas,"$2,914.31",14,0.13
43,East,District of Columbia,"$2,865.02",4,0.13
44,West,Wyoming,"$1,603.14",1,0.07
45,Central,South Dakota,"$1,315.56",5,0.06
46,East,Maine,"$1,270.53",3,0.06
47,East,West Virginia,"$1,209.82",2,0.05
48,Central,North Dakota,$919.91,2,0.04


-> State yang terbanyak dari jumlah order dan penjualan terdapat di State California dengan jumlah order 1.002, serta dengan nilai penjualan sebesar $446,306.46, kontribusi 19.73%

-> State dengan nominasi terkecil ada di state Nort Dakota jumlah order hanya 2, besar penjualan $919.91, dan kontribusi yang diberikan hanya  3.12%

7. Sales by City

In [54]:
city_sales = (
    df.groupby(["Region","State","City"]) ["Sales"]
    .sum()
    .sort_values(ascending=False)
    .reset_index()
)

city_order = (
    df.groupby(["Region","State","City"]) ["Order ID"]
    .nunique()
    .sort_values(ascending=False)
    .reset_index()
)

city_all = pd.merge(
    city_sales,
    city_order,
    on=["Region", "State", "City"]
)

# Menghitung kontribusi (%)
city_all["Contribution (%)"] = (
    city_all["Sales"] / city_all["Sales"].sum() * 100
).round(2)

city_all.head(10)

,Region,State,City,Sales,Order ID,Contribution (%)
0,East,New York,New York City,252462.5470,439,11.16
1,West,California,Los Angeles,173420.1810,378,7.67
2,West,Washington,Seattle,116106.3220,210,5.13
3,West,California,San Francisco,109041.1200,261,4.82
4,East,Pennsylvania,Philadelphia,108841.7490,262,4.81
5,Central,Texas,Houston,63956.1428,186,2.83
6,Central,Illinois,Chicago,47820.1330,166,2.11
7,West,California,San Diego,47521.0290,88,2.10
8,Central,Michigan,Detroit,42446.9440,53,1.88
9,South,Florida,Jacksonville,39133.3280,38,1.73


In [55]:
city_display = city_all.copy()

city_display["Sales"] = city_display["Sales"].map("${:,.2f}".format)

city_display.head(10)

,Region,State,City,Sales,Order ID,Contribution (%)
0,East,New York,New York City,"$252,462.55",439,11.16
1,West,California,Los Angeles,"$173,420.18",378,7.67
2,West,Washington,Seattle,"$116,106.32",210,5.13
3,West,California,San Francisco,"$109,041.12",261,4.82
4,East,Pennsylvania,Philadelphia,"$108,841.75",262,4.81
5,Central,Texas,Houston,"$63,956.14",186,2.83
6,Central,Illinois,Chicago,"$47,820.13",166,2.11
7,West,California,San Diego,"$47,521.03",88,2.10
8,Central,Michigan,Detroit,"$42,446.94",53,1.88
9,South,Florida,Jacksonville,"$39,133.33",38,1.73


In [56]:
city_display.tail(10)

,Region,State,City,Sales,Order ID,Contribution (%)
590,West,Utah,Layton,$4.96,1,0.0
591,South,Arkansas,Springdale,$4.30,1,0.0
592,West,Washington,Auburn,$4.18,1,0.0
593,West,Washington,Everett,$3.86,1,0.0
594,West,California,San Luis Obispo,$3.62,1,0.0
595,South,Florida,Ormond Beach,$2.81,1,0.0
596,South,Florida,Pensacola,$2.21,1,0.0
597,South,Florida,Jupiter,$2.06,1,0.0
598,East,Ohio,Elyria,$1.82,1,0.0
599,Central,Texas,Abilene,$1.39,1,0.0


-> New York merupakan Kota yang menyumbang penjualan terbanyak dengan nilai $252,462.55, memiliki order 439 paling besar diantara lainnya dan memiliki kontribusi sebesar 11.16%

-> Sebaliknya Kota bernama Abilene menyumbang penjualan sebesar $1.39 dengan order sebanyak 1 kali saja, banyak yg memiliki order hanya sekali tetapi nilai sales diatas dari kota Abilene

8. CONCLUSION

Region West memberikan kontribusi terbesar terhadap penjualan perusahaan,
memiliki jumlah order tertinggi serta customer terbanyak.
Hal ini menunjukkan bahwa wilayah tersebut merupakan pasar utama perusahaan.

Sebaliknya South memiliki kontribusi paling rendah sehingga perusahaan dapat
mengevaluasi strategi pemasaran maupun distribusi produk pada wilayah tersebut.